### PR TITLE
[TRB-17393]:  add util builders for SLO scale merge acton policy DTO

### PR DESCRIPTION
**Pull request by libai98**
This PR is update the turbo-go-sdk with the protobuf change related to SLO scale merge action: https://github.ibm.com/turbonomic/opsmgr/pull/21

1) Updated the go bindings for ActionMergePolicyDTO. These DTOs are to be sent by the probes during probe registration process

2) Added a builder to help go probes conveniently build the merge policy DTOs

### CODE REVIEW

File: pkg/builder/action_merge_meta_data_builder.go [hunk 1]
```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
...
+
+
+
+// Horizontal Scale Merge Policy DTO builder
+type HorizontalScaleMergePolicyBuilder struct {
+	entityType                *proto.EntityDTO_EntityType
+	aggregationTargets        []*ActionAggregationTargetBuilder
+	chainedAggregationTargets []*ActionDeDuplicateAndAggregationTargetBuilder
+	commTypes                 []*CommodityMergeData
+	entityFilters              []*proto.EntityDTO_WorkloadControllerData
```
Comment (amd-ibm) [inline pkg/builder/action_merge_meta_data_builder.go:200]: nit: formatting

File: pkg/builder/action_merge_meta_data_builder.go [hunk 2]
```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
...
+	hsb.aggregationTargets = append(hsb.aggregationTargets, mergeTarget)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) DeDuplicateAndAggregateBy(mergeTarget *ActionDeDuplicateAndAggregationTargetBuilder) *HorizontalScaleMergePolicyBuilder {
+	hsb.chainedAggregationTargets = append(hsb.chainedAggregationTargets, mergeTarget)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForWorkloadControllerDataFilter(controllerData proto.EntityDTO_WorkloadControllerData) *HorizontalScaleMergePolicyBuilder  {
```
Comment (amd-ibm) [inline pkg/builder/action_merge_meta_data_builder.go:222]: Shouldn't this be a `ContainerPodData` object? I thought we originally had `WorkloadControllerData` in the proto, but changed it to `ContainerPodData` after our discussions.

File: pkg/builder/action_merge_meta_data_builder.go [hunk 3]
```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
...
 	return mergeSpec, nil
 }
+
+
+
+
+// Horizontal Scale Merge Policy DTO builder
+type HorizontalScaleMergePolicyBuilder struct {
+	entityType                *proto.EntityDTO_EntityType
+	aggregationTargets        []*ActionAggregationTargetBuilder
```
Comment (amd-ibm) [inline pkg/builder/action_merge_meta_data_builder.go:197]: I don't believe that we need the `aggregationTargets`.

File: pkg/builder/action_merge_meta_data_builder.go [hunk 4]
```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
...
+	return hsb
+}
+
+// Create the ActionMergePolicyDTO for Horizontal Scale actions.
+func (hsb *HorizontalScaleMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error) {
+	if hsb.entityType == nil {
+		return nil, fmt.Errorf("Entity type required for horizontal scale merge policy")
+	}
+
+	if len(hsb.aggregationTargets) == 0 && len(hsb.chainedAggregationTargets) == 0 {
```
Comment (amd-ibm) [inline pkg/builder/action_merge_meta_data_builder.go:251]: Are you sure we need the `aggregationTargets` and that they should be required?

File: pkg/builder/action_merge_meta_data_builder.go [hunk 5]
```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
...
+	comm := &CommodityMergeData{
+		commType:    commType,
+		changedAttr: changedAttr,
+	}
+	hsb.commTypes = append(hsb.commTypes, comm)
+	return hsb
+}
+
+// Create the ActionMergePolicyDTO for Horizontal Scale actions.
+func (hsb *HorizontalScaleMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error) {
```
Comment (amd-ibm) [inline pkg/builder/action_merge_meta_data_builder.go:246]: Looks like you're missing the logic to set/pass the entity filters.

**Review by amd-ibm — CHANGES_REQUESTED** [review]

File: pkg/builder/action_merge_meta_data_builder.go (hunk 3 above)
Comment (libai98) [reply pkg/builder/action_merge_meta_data_builder.go]: Yes, I removed that. Thanks!

File: pkg/builder/action_merge_meta_data_builder.go (hunk 4 above)
Comment (libai98) [reply pkg/builder/action_merge_meta_data_builder.go]: removed as commented above. Thanks!

File: pkg/builder/action_merge_meta_data_builder.go (hunk 2 above)
Comment (libai98) [reply pkg/builder/action_merge_meta_data_builder.go]: Yes. Thanks!

**Pull request closed without merging**